exporter/syslogexporter/syslog.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2023, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syslogexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/syslogexporter"
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const defaultPriority = 165
+const defaultFacility = 1
+const versionRFC5424 = 1
+
+const formatRFC5424Str = "rfc5424"
+const formatRFC3164Str = "rfc3164"
+
+const priority = "priority"
+const facility = "facility"
+const version = "version"
+const hostname = "hostname"
+const app = "appname"
+const pid = "proc_id"
+const msgID = "msg_id"
+const structuredData = "structured_data"
+const message = "message"
+
+const emptyValue = "-"
+
+type Syslog struct {
```

Comment:
Why is this named `Syslog`? It seems a name like `sender` or `syslogSender` would be a better name, as it is concerned with sending the syslog data to a remote endpoint?

Also I don't think this struct needs to be exported (i.e. start with a capital letter)?